python/pyspark/sql/readwriter.py

```diff
@@ -431,6 +432,8 @@ def jdbc(self, url, table, column=None, lowerBound=None, upperBound=None, numPar
         if column is not None:
             if numPartitions is None:
                 numPartitions = self._spark._sc.defaultParallelism
```

Comment:
I think we should make the Scala API and Python API consistent. The existing Python API is not following [the document](http://spark.apache.org/docs/2.1.0/sql-programming-guide.html). 
```
These options must all be specified if any of them is specified. They describe how to partition the table when reading in parallel from multiple workers. partitionColumn must be a numeric column from the table in question. Notice that lowerBound and upperBound are just used to decide the partition stride, not for filtering the rows in table. So all rows in the table will be partitioned and returned. This option applies only to reading.
```